qiskit_experiments/calibration/calibrations.py

```diff
@@ -0,0 +1,1025 @@
+                    )
+                )
+            if isinstance(chan, (DriveChannel, MeasureChannel)):
+                qubit_set.add(chan.index)
+
+            if isinstance(chan, ControlChannel):
+                for qubit in self._controls_config_r[chan]:
+                    qubit_set.add(qubit)
+
+        qubits_ = tuple(qubit for qubit in qubits if qubit in qubit_set)
+
+        # Complete the assignment dictionary with any missing parameter couplings
+        assign_params_ = dict(assign_params)
+        for param_key, value in assign_params.items():
+            # Iterate over all keys that point to the parameter pointed to by the
+            # the key param_key.
+            for key in self._parameter_map_r[self.calibration_parameter(*param_key)]:
+                if (
+                    ParameterKey(key.parameter, qubits_, key.schedule) not in assign_params
+                    and key.qubits == qubits_
+                ):
+                    assign_params_[ParameterKey(key.parameter, qubits_, key.schedule)] = value
+
+        assign_params = assign_params_
+
+        # 2) Recursively assign the parameters in the called instructions.
+        ret_schedule = ScheduleBlock(
+            alignment_context=schedule.alignment_context,
+            name=schedule.name,
+            metadata=schedule.metadata,
+        )
+
+        for inst in schedule.blocks:
+            if isinstance(inst, Call):
+                inst = inst.assigned_subroutine()
+
+            if isinstance(inst, ScheduleBlock):
+                inst = self._assign(inst, qubits_, assign_params, group, cutoff_date)
+
+            ret_schedule.append(inst, inplace=True)
+
+        # 3) Get the parameter keys of the remaining instructions. At this point in
+        #    _assign all parameters in Call instructions that are supposed to be
+        #     assigned have been assigned.
+        keys = set()
+
+        if ret_schedule.name in set(key.schedule for key in self._parameter_map):
+            for param in ret_schedule.parameters:
+                keys.add(ParameterKey(param.name, qubits_, ret_schedule.name))
+
+        # 4) Build the parameter binding dictionary.
+        assign_params = assign_params if assign_params else dict()
+
+        binding_dict = {}
+        for key in keys:
+            # Get the parameter object. Since we are dealing with a schedule the name of
+            # the schedule is always defined. However, the parameter may be a default
+            # parameter for all qubits, i.e. qubits may be an empty tuple.
+            if key in self._parameter_map:
+                param = self._parameter_map[key]
+            elif ParameterKey(key.parameter, (), key.schedule) in self._parameter_map:
+                param = self._parameter_map[ParameterKey(key.parameter, (), key.schedule)]
+            else:
+                raise CalibrationError(f"Parameter key {key} has no parameter.")
```

Comment:
I feel like I asked this about a similar spot in the code before, but could this become `param = self.calibration_parameter(*key)`?

Reply:
Good point. See here c597e52